torch/_dynamo/symbolic_convert.py

```diff
@@ -952,40 +946,35 @@ def COMPARE_OP(self, inst):
                     **options,
                 )
             )
-        elif (
-            isinstance(left, TensorVariable) or isinstance(right, TensorVariable)
-        ) and op in supported_tensors:
-            self.push(
-                wrap_fx_proxy(
-                    self,
-                    supported_tensors[op](left.as_proxy(), right.as_proxy()),
-                    **options,
-                )
-            )
-        elif (
-            isinstance(left, DynamicShapeVariable)
-            or isinstance(right, DynamicShapeVariable)
-        ) and op in supported_tensors:
-            self.push(
-                DynamicShapeVariable.create(
-                    self,
-                    supported_tensors[op](left.as_proxy(), right.as_proxy()),
-                    dyn_shape=None,
-                    **options,
-                )
-            )
         elif op in ("in", "not in"):
             self.push(right.call_method(self, "__contains__", [left], {}))
             if op == "not in":
                 self.UNARY_NOT(inst)
         elif (
-            isinstance(left, UserFunctionVariable)
-            and isinstance(right, UserFunctionVariable)
-            and op in supported_is_const
-        ):
-            self.push(
-                ConstantVariable(supported_is_const[op](left.fn, right.fn), **options)
+            # TODO: feels like we should be able to refactor so that VariableTracker.compare()
+            # covers more cases. However some of the cases are simpler to leave in the current function.
+            # (for example, if both left an right are python constants, always return a constant
+            # instead of re-writing that logic for every type of VariableTracker that can be constant)
+            isinstance(
+                left,
+                (
+                    TensorVariable,
+                    DynamicShapeVariable,
+                    BaseListVariable,
+                    UserFunctionVariable,
+                ),
             )
+            or isinstance(
```

Comment:
This `or` seems a bit inconsistent.  If `left` fails the check we will call the base variable tracker `compare()` (that just raises the same error as the `else` down below).

Given we define a base class version of compare, we could just call this unconditionally.